Update which Agent services run as root/non-root (#12449)
DataDog/documentation: content/en/security/agent.md

## Running as an unprivileged user

> By default, the Agent runs as the `dd-agent` user on Linux and as the `ddagentuser` account on [Windows][18]. Note there are some exceptions to this: the `system-probe` which runs as `root` on Linux and as `LOCAL_SYSTEM` on Windows, the `process-agent` which runs as `LOCAL_SYSTEM` on Windows, and the `security-agent` which runs as `root` on Linux.

Reviewer: ```suggestion
By default, the Agent runs as the `dd-agent` user on Linux and as the `ddagentuser` account on [Windows][18]. The exceptions are as follows:

- The `system-probe` runs as `root` on Linux and as `LOCAL_SYSTEM` on Windows.
- The `process-agent` runs as `LOCAL_SYSTEM` on Windows.
- The `security-agent` runs as `root` on Linux.
```

## Running as an unprivileged user

By default, the Agent runs as the `dd-agent` user on Linux and as the `ddagentuser` account on [Windows][18]. Note there are some exceptions to this: the `system-probe` which runs as `root` on Linux and as `LOCAL_SYSTEM` on Windows, the `process-agent` which runs as `LOCAL_SYSTEM` on Windows, and the `security-agent` which runs as `root` on Linux.

> The `process-agent` can be configured to run as the unprivileged `ddagentuser` on Windows, but some functionality will be missing. In specific, the command line of each process won't be available in Datadog.

Reviewer 2: I don't know that this is true.
I mean, we could in theory do that, but AFAIK there's nothing in the install to do this, and the result will be very continuous errors (there's nothing in the code to detect we're _not_ running as root so the various api calls would throw errors all over) .  Again, AFAIK.

Reviewer 3: Sounds like we need to confirm or correct this before adding it to the docs.

Reviewer 4: Is there a reason we want to state this particular exception? TBH I don't _quite_ remember what the effect is, but if the command line is not available then the process can't be reasonably ingested by our pipeline, so it's either dropped at by the agent or very early in our pipeline. Why not just leave it be and "force" the customer to run with privileges?

Author: I removed that line.
